File: sortingAlgorithms/day16/quickSort.cpp
#include <iostream>
#include <vector>

using namespace std;

int partition(vector<int> &arr, int left, int right)
{
    int pivot = arr[right];
    int store_index = left - 1;
    for (int i = left; i < right; i++)
    {
        if (arr[i] < arr[pivot])
        {
            store_index++;
            swap(arr[store_index], arr[i]);
        }
    }

    store_index++;
    swap(arr[store_index], arr[right]);

    return store_index;
}

void quickSort(vector<int> &arr, int left, int right)
{
    if (left < right)
    {
        int pivot = partition(arr, left, right);
        quickSort(arr, left, pivot - 1);
        quickSort(arr, pivot + 1, right);
    }
}

File: sortingAlgorithms/day18/quickSort.cs
using System;
using System.Collections.Generic;

class Program
{
    static int partition(int[] arr, int left, int right)
    {
        int pivot = arr[right];
        int store_index = left - 1;

        for (int i = left; i < right; i++)
        {
            if (arr[i] < pivot)
            {
                store_index++;
                int temp = arr[store_index];
                arr[store_index] = arr[i];
                arr[i] = temp;
            }
        }
        store_index++;



    }
}